pandas/io/tests/test_parsers.py

```diff
@@ -2342,6 +2342,20 @@ def test_usecols(self):
         expected = expected[['a', 'b']]
         tm.assert_frame_equal(result, expected)
 
+        # 11823: usecols vs no usecols
+        result = self.read_csv(StringIO(data), names=['a', 'a', 'b'],
+                               header=None, usecols=['a', 'a', 'b'])
+        expected = self.read_csv(StringIO(data), names=['a', 'a', 'b'],
+                                 header=None)
```

Comment:
I'd like to use a fixed result, here IOW, actually construct the expected ``DataFrame``, rather than reading it in with a different set of options.